### PR TITLE
Skip slow checkpointing tests when sanitizer is enabled

### PR DESCRIPTION
**Pull request by szkarpinski**
## Category:

**Other** - tests optimization

## Description:

In this PR I disable slow external source tests, stateless operators tests and Pytorch tests if sanitizer is enabled.

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: test_dali_tf_plugin.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from nvidia.dali import pipeline_def
```
Comment (JanuszL) [inline test_dali_tf_plugin.py:1]: Leftover?

Comment (szkarpinski) [reply test_dali_tf_plugin.py]: Oops, removed

**Review by jantonguirao — APPROVED** [review]

File: qa/TL0_python-self-test-core/test_body.sh [hunk 2]
```diff
@@ -40,14 +40,23 @@ test_type_annotations() {
...
-    ${python_new_invoke_test} -A '!slow,!pytorch,!mxnet,!cupy,!numba' checkpointing.test_dali_checkpointing
-    ${python_new_invoke_test} -A '!slow,!pytorch,!mxnet,!cupy,!numba' checkpointing.test_dali_stateless_operators
+    if [ -z "$DALI_ENABLE_SANITIZERS" ]; then
+        ${python_new_invoke_test} -A '!slow,!pytorch,!mxnet,!cupy,!numba' checkpointing.test_dali_checkpointing
+        ${python_new_invoke_test} -A '!slow,!pytorch,!mxnet,!cupy,!numba' checkpointing.test_dali_stateless_operators
+    else
+        ${python_new_invoke_test} -A '!slow,!pytorch,!mxnet,!cupy,!numba,!sanitizer_skip' checkpointing.test_dali_checkpointing
+
+        # External source tests are slow and Python-side mostly, but let's run just one of them
+        ${python_new_invoke_test} -A '!slow,!pytorch,!mxnet,!cupy,!numba,!san_skip' checkpointing.test_dali_checkpointing.test_external_source_checkpointing:1
```
Comment (stiepan) [inline qa/TL0_python-self-test-core/test_body.sh:58]: Do we use `san_skip` anywhere?

Comment (szkarpinski) [reply qa/TL0_python-self-test-core/test_body.sh:58]: Oops! It's a mistake (or two mistakes): 1) we don't use san_skip, we use sanitizer_skip 2) here we *don't* want to skip the test. Thanks!

Comment (szkarpinski) [reply qa/TL0_python-self-test-core/test_body.sh]: Removed

**Comment by szkarpinski** [comment]
!build

**Pull request merged**